Library/Homebrew/extend/ARGV.rb

```diff
@@ -25,7 +25,7 @@ def formulae
   def resolved_formulae
     require "formula"
     @resolved_formulae ||= (downcased_unique_named - casks).map do |name|
-      if name.include?("/")
+      if name.include?("/") || File.exist?(name)
```

Comment:
The issue tried to solve is as following. You have a custom formula `foo.rb`. You should be able to do thing like `brew install/reinstall foo.rb`. And it should be loading the custom formula. 

You are right we probably should check file ext.